paradox-nakadi-consumer-core/src/main/java/de/zalando/paradox/nakadi/consumer/core/FailedEventSource.java

```diff
@@ -0,0 +1,16 @@
+package de.zalando.paradox.nakadi.consumer.core;
+
+import java.util.Optional;
+
+import de.zalando.paradox.nakadi.consumer.core.domain.FailedEvent;
+
+public interface FailedEventSource<T extends FailedEvent> {
+
+    Optional<T> getFailedEvent();
+
+    void commit(T t);
+
+    String getEventSourceName();
+
+    long getApproximatelyTotalNumberOfFailedEvents();
```

Comment:
I think this method name is superfluous. The fact that the number of failed events is an approximate and can change is an implementation detail. IMO something like `getSize` is enough.